examples/DotNetCore/BackupStore/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,146 @@
+
+                // Connect to Azure App Configuration stores
+                ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+                ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+
+                // Connect to storage queue
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(Environment.GetEnvironmentVariable("StorageQueueConnectionString"));
+                CloudQueueClient queueClient = storageAccount.CreateCloudQueueClient();
+                CloudQueue queue = queueClient.GetQueueReference("eventgridqueue");
+                CloudQueueMessage customMessage = new CloudQueueMessage($"Refreshing Azure App Configuration store at: {DateTime.Now}.");
+
+                // Peek at the next message
+                CloudQueueMessage peekedMessage = queue.PeekMessage();
+                while (peekedMessage != null)
+                {
+                    var messageBody = peekedMessage.AsString;
+                    try
+                    {
+                        JObject dataObject = JObject.Parse(messageBody);
+                        string topic = dataObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
+                        {
+                            // We have events that need to be processed
+                            // Add custom message to the end of queue
+                            queue.AddMessage(customMessage);
+                            break;
+                        }
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // If its not a valid JSON, ignore the message 
+                        // because it will be discarded from the queue
+                    }
+                    log.LogInformation($"Deleting message from queue: {messageBody}");
+                    queue.DeleteMessage(queue.GetMessage());
+                    peekedMessage = queue.PeekMessage();
+                }
+
+                // return if there are no messages in the queue
+                if (peekedMessage == null)
+                {
+                    log.LogInformation($"No events received for processing.");
+                    return;
+                }
+
+                // Read all settings from Primary Store and create local copy
+                IEnumerable<ConfigurationSetting> primaryStoreSettings = primaryAppConfigClient.GetConfigurationSettings(new SettingSelector());
+                IDictionary<Tuple<string, string>, ConfigurationSetting> primaryStoreLocalSettings = new Dictionary<Tuple<string, string>, ConfigurationSetting>();
```

Comment:
Could we use a custom type instead of a `Tuple` to make it clear what the individual strings are?

Reply:
done